backend/inspirehep/celery.py

```diff
@@ -21,16 +23,41 @@
 
 
 class CeleryTask(AppContextTask):
-
     def on_failure(self, exc, task_id, args, kwargs, einfo):
-        if isinstance(exc, (InvalidRequestError, OperationalError, Psycopg2OperationalError)):
-            LOGGER.exception('Shutting down celery process')
+        if isinstance(
+            exc, (InvalidRequestError, OperationalError, Psycopg2OperationalError)
+        ):
+            LOGGER.exception("Shutting down celery process")
             try:
-                with open('/dev/termination-log', 'w') as term_log:
+                with open("/dev/termination-log", "w") as term_log:
                     term_log.write(str(exc))
             finally:
                 os.kill(os.getppid(), signal.SIGTERM)
 
 
 celery = create_celery_app(create_app(LOGGING_SENTRY_CELERY=True))
 celery.Task = CeleryTask
+
+
+@worker_process_init.connect()
+def setup_metrics_server(*args, **kwargs):
+    """Expose prometheus metrics.
+
+    Exposes prometheus metrics for worker through prometheus web server on specified port.
+    This function assumes that on production there is concurrency set to 1,
+    otherwise it will expose metrics only form first worker child.
+    We are outside flask app context here, so there is no access to config.
+    """
+    port = os.getenv("PROMETHEUS_METRICS_PORT", 9090)
+    host = os.getenv("PROMETHEUS_METRICS_HOST", "")
+    try:
+        start_http_server(port, host)
+        LOGGER.info(message="Exposing prometheus metrics.", port=port)
+    except OSError:
+        # Other (than the first one) celery children
+        # will fail starting server for metrics
+        LOGGER.warning(
+            "Cannot use default port for serving prometheus metrics. "
+            "This should happen only on local machine. "
+            "If you have this on production then you have a problem."
```

Comment:
```suggestion
            "Cannot serve prometheus metrics from this worker, probably because concurrency >= 1 and another worker has already exposed metrics. This shouldn't happen in production!"
```
Also add `host` and `port`?

Reply:
:+1: